frontend/src/models/models.js

```diff
@@ -89,6 +89,37 @@ module.exports = app => app.component('models', {
     this.status = 'loaded';
   },
   methods: {
+    clickFilter(path) {
+      if (this.searchText) {
+        let searchObj;
+      
+        try {
+          // Try to parse this.searchText if it's a valid JSON string
+          searchObj = JSON.parse(this.searchText);
+        } catch (e) {
+          // If it fails, default to an empty object
+          console.log('could not add to filter', e);
```

Comment:
Except it doesn't default to an empty object here, `searchObj` is undefined if `JSON.parse()` throws an error